cmd/rehearse/main.go

```diff
@@ -0,0 +1,222 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+
+	// TODO: Solve this properly
+	// "github.com/davecgh/go-spew/spew"
+	"github.com/getlantern/deepcopy"
+	// "github.com/ghodss/yaml"
+	"github.com/sirupsen/logrus"
+
+	pjapi "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	pjclientset "k8s.io/test-infra/prow/client/clientset/versioned"
+	"k8s.io/test-infra/prow/config"
+	prowflagutil "k8s.io/test-infra/prow/flagutil"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pjutil"
+
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func getJobsToExecute(ca *config.Agent) config.JobConfig {
+	var youHaveOneJob config.Presubmit
+	for _, job := range ca.Config().Presubmits["openshift/ci-operator"] {
+		if job.Name == "pull-ci-openshift-ci-operator-master-build" {
+			youHaveOneJob = job
+			break
+		}
+	}
+
+	jobs := config.JobConfig{
+		Presubmits: map[string][]config.Presubmit{"openshift/ci-operator": []config.Presubmit{youHaveOneJob}},
+	}
+	return jobs
+}
+
+func makeRehearsalPresubmit(source *config.Presubmit, repo string, prNumber int) (*config.Presubmit, error) {
+	var rehearsal config.Presubmit
+	deepcopy.Copy(&rehearsal, source)
+
+	rehearsal.Name = fmt.Sprintf("rehearse-%d-%s", prNumber, source.Name)
+	rehearsal.Context = fmt.Sprintf("ci/rehearse/%s/%s", repo, strings.TrimPrefix(source.Context, "ci/prow/"))
+
+	if len(source.Spec.Containers) != 1 {
+		return nil, fmt.Errorf("Cannot rehearse jobs with more than 1 container in Spec")
+	}
+	container := source.Spec.Containers[0]
+
+	if len(container.Command) != 1 || container.Command[0] != "ci-operator" {
+		return nil, fmt.Errorf("Cannot rehearse jobs that have Command different from simple 'ci-operator'")
+	}
+
+	for _, arg := range container.Args {
+		if strings.HasPrefix(arg, "--git-ref") || strings.HasPrefix(arg, "-git-ref") {
+			return nil, fmt.Errorf("Cannot rehearse jobs that call ci-operator with '--git-ref' arg")
+		}
+	}
+
+	if len(source.Branches) != 1 {
+		return nil, fmt.Errorf("Cannot rehearse jobs that run over multiple branches")
+	}
+	branch := strings.TrimPrefix(strings.TrimSuffix(source.Branches[0], "$"), "^")
+
+	gitrefArg := fmt.Sprintf("--git-ref=%s@%s", repo, branch)
```

Comment:
Wait, why aren't we setting the `ProwJob.Spec.Refs` ?

Reply:
Because I do not know the full impact of messing with these `ProwJob` fields. Wouldn't it break the attachment of these jobs to the `openshift/release` PRs? I don't know. If setting `ProwJob.Spec.Refs` is only used to pass the `$JOB_SPEC` to the job, then setting it is really they way to do it, thanks!

Reply:
We should be able to set the `Spec.Refs` to change the _main_ refs used for the test, which will be the ones that get used for reporting etc. We can use `extra_refs` to set other things we want to clone, like the repo under test. But not sure, that would mess up the dir that things get cloned into. Hrm.

Reply:
I vote to keep the `--git-ref` way for now to get the feature working and if needed, we can optimize the internals later.